### ISSUE TITLE
Add the possibility of disabling facet values in the UI instead of removing them.

### ISSUE THREAD

**Issue by seafoox** [algolia/instantsearch.js#840]
By design, the API only returns the facets value available for the current search context. That easily allows developers to only display the facet values that are available, and not the other. Which is very handy when the set of potential values is large, but which can also be less legible in some UX.

The idea here would be to be able to keep those facet values displayed all the time, make them look as `disabled` when those are not available for the current search context. 

That would I guess be applicable to both `menu` and `refinementList` widgets.

**Comment by vvo** [comment]
What would be a good API option name for this?

**Comment by loryhuz** [comment]
I've the same problem, not with instantsearch.js, but on native mobile platforms. But it's a feature needed on all platforms I guess.

I've implemented faceting on e-commerce app, and there is a lot of combinations that's returns 0 products. So I want to disable facets that's returns 0 products to improve UX of this feature.

![filters](https://cloud.githubusercontent.com/assets/3198863/15958503/4f87d456-2ef6-11e6-8971-c9ebaf6d94e4.png)

For example if I select filter events, it will return only red and yellow colors. Cool, by keeping the original facets lists, I can easily disable others colors (I mean, they will stay on the screen but with disabled state).

Problem is that theme facets will return only the facets I selected (seems legit). But then it's hard to handle on our side when there is multiple combinations of facets in different categories.
Would be cool to have an API option to have enabled/disabled state for each facet in the API response.
